### ISSUE TITLE
Morris: Preserve manager deliverycharge override at order split

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2238]
_Imported from trac ticket 2238.
Created by **dwarren**
Opened in trac: Mar 20, 2013, 3:20:30 PM
Last modified in trac: Aug 18, 2016, 1:31:13 PM_
***
_by dwarren:_
Scott,

I know we talked about the problem with delivery fees in Loft.  If someone changes the delivery fee and then later splits the sale, the delivery fee is changed back to the full amount.  Do we have a ticket open to make this not happen?

Darlene

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 20, 2013, 4:06:06 PM:_

Loft currently is programmed with the notion that it is not safe to assume that a split order should only charge a discounted delivery charge.  In other words, if the order is split, thus requiring more than one trip to the customer's home, a manager may say "well, now, in this case, we can't offer the discounted delivery charge anymore."

It sounds like Morris does not want to take that conservative approach, but rather would like to honor the manager's discounted delivery charge even if the order is split multiple times.  Is that an accurate statement?

(If that is accurate, I seem to recall some technical reasons this very difficult to honor, but don't remember immediately why.  I'm not 100% sure we can accommodate this...)

In any case, let me know,
Thanks,
Kent

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 21, 2013, 9:19:48 AM:_

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 21, 2013, 9:29:28 AM:_

The suggestion other retailers of our would make is leave it the way it is so that if the delivery picture changes, a manager needs to "re-approve" the discounted charge.

However, if that is not according to Morris's business rules, we can appreciate that and try to work with you... my fear is that I recall technical issues with this, but I don't remember what they are. This is extremely (horribly) complex code because Loft is attempting to, at the retailer's option, charge only once for the entire "Customer Order" (main order and all sibling split tickets), calculating what the "right" charge is if two tickets happen to have differing charges (use the highest in that case), while moving the charge to (and keeping it at) the most appropriate ticket (the first to be delivered), all while attempting to honor manager overrides, if any had been made.

More than a complaint, my point is I'd need to invest a good chunk of time to figure out what implications exist if we try to change how it's working, so I just want to make sure you want the change before I start (other retailers saw Loft's current behavior as valuable).

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 21, 2013, 9:29:39 AM:_

The order number in TRAIN is order 0305300AJ45.  Currently the discounted delivery fee is $49.95 with nothing split.

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 22, 2013, 8:46:18 AM:_

>What if we create a rule that the delivery fee would recalculate only if the Shipping (Zone) was changed ?  Would that be easier? 


Unfortunately, no, it wouldn't help simplify._[snip]_

**Comment by kentbower** [comment]
_**kbower** trac comment on Mar 22, 2013, 8:48:04 AM:_

Waiting for Morris to answer definitely they need this, see comment:5

**Comment by kentbower** [comment]
_**kbower** trac comment on Aug 18, 2016, 12:38:38 PM:_

Sameer, you do this already?

**Comment by kentbower** [comment]
_**kbower** trac comment on Aug 18, 2016, 1:31:12 PM:_

See [#2604](2604) (and also [#3501](3501))

**Comment by kentbower** [comment]
_**kbower** trac status change on Aug 18, 2016, 1:31:12 PM:_ **closed**

**Comment by kentbower** [comment]
_**kbower** trac comment on Aug 18, 2016, 1:31:13 PM:_

Closed as duplicate of [#2604](2604).
